lib/phoenix/sync/write.ex

```diff
@@ -0,0 +1,970 @@
+defmodule Phoenix.Sync.Write do
+  @moduledoc """
+  Provides [optimistic write](https://electric-sql.com/docs/guides/writes)
+  support for Phoenix- or Plug-based apps.
+
+  The client uses some local database and syncs changes into this database from
+  the server using [Electric](https://electric-sql.com/).
+
+  Local writes are performed on the local database using
```

Comment:
I think it's important decouple the write handling in Phoenix from the use of TanStack/optimistic. Instead, what we can do is define the data structure that this handles. It's then possible for any front-end system that's building up a set of local changes to send this structure over the wire. TanStack/optimistic is one implementation of a client side library that does this.

Ideally we want to extract the structure, call it something and document it standalone. With reference validator, like a JSON Schema or whatever.

Reply:
Wasn't sure about the exact relation between this and tanstack/optimistic. Essentially it is decoupled, but I think it's helpful to have some reference impl that makes it easy to try this out. 

The `Phoenix.Sync.Write.Mutation` module has some docs around the data structure that I can expand and link to.